packages/compartment-mapper/src/node-modules.js

```diff
@@ -125,6 +125,55 @@ const readDescriptorWithMemo = async (memo, maybeRead, packageLocation) => {
   return promise;
 };
 
+/**
+ *
+ * @param {Graph} graph
+ * @param {string} packageLocation
+ * @param {string[]} logicalPath
+ * @param {Map<string, string[]>} [preferredPackageLogicalPathMap]
+ * @returns {Promise<void>}
+ */
+const updateShortestPaths = async (
```

Comment:
My intuition is that this function should be synchronous since it only needs to visit the nodes that have already been discovered and assigned potentially-longer paths.